.travis.yml

```diff
@@ -1,33 +1,37 @@
+---
 language: python
 python:
-  - "3.6" # EOL 2021-12-23
-  - "3.7" # EOL 2023-06-27
-  - "3.8" # EOL 2024-10
-  - "3.9-dev" # RC1 2020-09-10
-  - "nightly"
+    - '3.6' # EOL 2021-12-23
+    - '3.7' # EOL 2023-06-27
+    - '3.8' # EOL 2024-10
+    - 3.9-dev # RC1 2020-09-10
+    - nightly
 matrix:
-  allow_failures:
-    - python: "3.9-dev"
-    - python: "nightly"
-  fast_finish: true
+    allow_failures:
+        - python: 3.9-dev
+        - python: nightly
+    fast_finish: true
+cache:
+    directores:
+        - $HOME/.cache/pre-commit
 
 install:
-  - "pip install -qU --compile pip"
-  - "pip install -r requirements.txt"
+    - pip install -qU --compile pip
+    - pip install -r requirements.txt
 
 script:
-  - "python setup.py sdist"
-  - "pip install ."
-  - "make check"
+    - python setup.py sdist
+    - pip install .
+    - pre-commit run --all-files
```

Comment:
From personal experience, consider using pre-commit as a GitHub action rather than as part of Travis CI.

It reduces the CI time, as it doesn't run with every single Python version separately, and the checks can run incrementally.

Reply:
Yep, I'm thinking about that transition but one step at a time.